hack/bazel-server.sh

```diff
@@ -1,5 +1,16 @@
-BAZEL_PID=$(bazel info | grep server_pid | cut -d " " -f 2)
-while kill -0 $BAZEL_PID 2>/dev/null; do sleep 1; done
-# Might not be necessary, just to be sure that exec shutdowns always succeed
-# and are not killed by docker.
-sleep 1
+ARCH=$(uname -m)
+
+# s390x builder image is not shipped with bazel.
+# Below code is causing the unit test failure in s390x.
+if [ $ARCH != "s390x" ]; then
+  BAZEL_PID=$(bazel info | grep server_pid | cut -d " " -f 2)
+  while kill -0 $BAZEL_PID 2>/dev/null; do sleep 1; done
+  # Might not be necessary, just to be sure that exec shutdowns always succeed
+  # and are not killed by docker.
+  sleep 1
+else
+  #Sleep for infinite time to run the s390x unit tests.
+   while true; do
```

Comment:
Why is this infinite sleep required?

Reply:
The script is called as the run command for the builder container. As such it is the "main" process, whereas the other steps are executed via `exec`. Therefore the container exits when the `hack/bazel-server.sh` script exits. An alternative to infinite sleep we could maybe do something like 4h or another sensible timeout?

Regardles, @vamsikrishna-siddu i would not use an endless loop here but rather call `sleep infinity` directly, to avoid wasting CPU cycles on the while loop.

Reply:
Hi @jschintag  i removed the endless loop and changed the command to sleep infinity